tests: sched: preempt: Fix the skipped testing when enabling SMP

yinfangw opened the pull request:
When SMP feature is enabled on board, this test suite will skip.
Enable the test suite by removing the filter.
Note that these testings only run SMP platform on single core
which are handled by setting MP_NUM_CPUS=1 in prj.conf.

evgeniy-paltsev commented on tests/kernel/sched/preempt/testcase.yaml:
```diff
@@ -1,8 +1,9 @@
-    filter: not CONFIG_SMP
     platform_exclude: nrf52dk_nrf52810
+    extra_configs:
+      - CONFIG_SMP=n
```
Hmm, as I remember we do not disable SMP from the test suit. If we need execution only on one core we set `CONFIG_MP_NUM_CPUS` to 1.

yinfangw replied:
> Hmm, as I remember we do not disable SMP from the test suit. If we need execution only on one core we set `CONFIG_MP_NUM_CPUS` to 1.

This original test suite works when CONFIG_SMP is not enabled. Currently when the board configure as CONFIG_SMP by default for all other testing, this testing case will be skipped. After I do these changes so that the board by default with CONFIG_SMP can run this case.

yinfangw replied:
> Hmm, as I remember we do not disable SMP from the test suit. If we need execution only on one core we set `CONFIG_MP_NUM_CPUS` to 1.

This original test suite works when CONFIG_SMP is not enabled. Currently when the board configure as CONFIG_SMP by default for all other testing, this testing case will be skipped. After I do these changes so that the board by default with CONFIG_SMP can run this case.

andyross replied:
Yeah, this filter is ancient and basically wrong in today's environment where we have lots of SMP platforms (which weren't getting coverage in this test!).  Agree we should remove it.

But also agreed: the "extra_configs" here isn't doing anything useful.  There's no need to set SMP=n, we just need to know there's only one cpu (which is done in prj.conf already).

yinfangw replied:
> Yeah, this filter is ancient and basically wrong in today's environment where we have lots of SMP platforms (which weren't getting coverage in this test!). Agree we should remove it.
> 
> But also agreed: the "extra_configs" here isn't doing anything useful. There's no need to set SMP=n, we just need to know there's only one cpu (which is done in prj.conf already).

Thank you Andy. I will fix as you said.

yinfangw committed:
tests: sched: preempt: Fix the skipped testing when enabling SMP

When SMP feature is enabled on board, this test suite will skip.
Enable the test suite by removing the filter.
Note that these testings only run SMP platform on single core
which are handled by setting MP_NUM_CPUS=1 in prj.conf.

Signed-off-by: Yinfang Wang <yinfang.wang@intel.com>

yinfangw commented:
@evgeniy-paltsev @andyross please help to review this patch.

evgeniy-paltsev reviewed (changes requested):
Hmm,

are you really sure that this test is ready for execution on SMP?

I briefly check the `tests/kernel/sched/preempt` test - we create multiple 'worker' threads which accesses (reads an writes) global variables like `last_thread` without any locking. That doesn't look like a thing which ready for SMP.

andyross commented:
Coming by, noticed this had a -1: @evgeniy-paltsev this looks OK.  In fact the preempt test is **not** MP-safe, by design.  (It's creating conditions under which it knows the scheduler will need to preempt threads and verifying it makes the right decision -- really it's sort of a too-clever-for-its-own-good kind of test).

But that's handled by setting MP_NUM_CPUS=1 in prj.conf.  The bug being addressed here is that there was also a filter that skipped the test on platforms with SMP=y, which meant they weren't getting coverage even though the test would be able to run just fine.

evgeniy-paltsev commented:
Ok, I see now. Unfortunately it's impossible to understand that looking on this commit/PR message and changes.

@yinfangw may I ask you to extend the commit & PR message with the info that we don't run the test on multiple cores, but only allow it to run on SMP platforms on single core?

Thanks!
